src/Pelagos/Bundle/AppBundle/Resources/public/js/sidebyside.js

```diff
@@ -96,6 +106,131 @@ $(document).ready(function()
             $(".filetabs", this).tabs();
+            rightLoaded = true;
+            loading.notify();
         });
     });
-});
\ No newline at end of file
+
+    $("#show-diff-button").button("option", "disabled", true).click(function (){
+        showDifferences();
+    });
+
+    loading.progress(function() {
+        if (leftLoaded && rightLoaded) {
+            $("#show-diff-button").button("enable");
+        } else {
+            $("#show-diff-button").button("disable");
+        }
+    });
+
+    function showDifferences()
+    {
+        // Change textarea and input to divs that look like them.
+        $("#right")
+        .find("textarea,input[type=text]")
+        .not(".keywordinput")
+        .each(function() {
+            var originalInput = $(this);
+            var inputType = originalInput.prop("tagName");
+            var inputHeight = originalInput.height();
+            var inputWidth = originalInput.width();
+            var inputText = originalInput.val();
+            var attrs = this.attributes;
+            var newElement = $("<div>")
+            .attr("tagname", inputType)
+            .height(inputHeight)
+            .width(inputWidth)
+            .text(inputText);
+
+            // Set all the attributes (name, id, etc).
+            $.each(attrs, function(index, atribute) {
+                newElement.attr(atribute.name, atribute.value);
+            });
+
+            newElement.addClass("contentbox");
+            originalInput.replaceWith(newElement);
+        });
+
+        $("#right").find(".contentbox")
+        .each(function(){
+            var thisId = $(this).attr("id");
+            var rightText = $(this).text();
+            var leftInput = $("#left").find("#"+thisId);
+            var leftText = leftInput.val();
+
+            // Comparing Text
+            var dmp = new diff_match_patch();
+            var d = dmp.diff_main(leftText, rightText);
+            dmp.diff_cleanupSemantic(d);
+            var diffHTML =  dmp.diff_prettyHtml(d);
+
+            // Setting the HTML with diff.
+            $(this).html(diffHTML);
+        });
+
+        // Compare keyword selects.
+        $("#right").find("select.keywordinput").each(function() {
+            var thisId = $(this).attr("id");
+            var leftSelect = $("#left").find("#"+thisId);
+            var rightSelect = $(this);
+
+            // Convert option list to lines.
+            var leftSelectText = leftSelect
+            .find("option")
+            .map(function(){return this.text;})
+            .get().join("\n");
+            var rightSelectText = rightSelect
+            .find("option")
+            .map(function(){return this.text;})
+            .get().join("\n");
+
+            // Compare in Linemode, put each option on a new line.
+            var dmp = new diff_match_patch();
+            var a = dmp.diff_linesToChars_(leftSelectText, rightSelectText);
+            var lineText1 = a.chars1;
```

Comment:
Used only once, not required to create a variable for this. Similarly for lineText2 and LineArray